tsdb/mapper.go

```diff
@@ -428,6 +447,36 @@ func (a *aggTagSetCursor) Next() (time int64, value interface{}) {
 	return a.nextFunc()
 }
 
+type rawPoint struct {
+	timestamp int64
+	value     []byte
+	cursor    *seriesCursor // cursor whence rawPoint came
+}
```

Comment:
The name `rawPoints` and `rawPoint` seem very generic. Maybe `pointHeap` and `pointHeapItem`? It'd also be nice to move the initialization from the cursors into a `newPointHeap()` function and move the reading and fetching into a `Next()` function on the `pointHeap`. That way the heap is more encapsulated.